server/form-pages/apply/reasons-for-placement/basic-information/exceptionDetails.ts

```diff
@@ -0,0 +1,101 @@
+import type { ObjectWithDateParts, TaskListErrors, YesOrNo } from '@approved-premises/ui'
+import { sentenceCase } from '../../../../utils/utils'
+import { Page } from '../../../utils/decorators'
+import { dateAndTimeInputsAreValidDates, DateFormats } from '../../../../utils/dateUtils'
+
+import TasklistPage from '../../../tasklistPage'
+
+export type ExceptionDetailsBody = ObjectWithDateParts<'agreementDate'> & {
+  agreedCaseWithManager: YesOrNo
+  managerName: string
+  agreementSummary: string
+}
+
+@Page({
+  name: 'exception-details',
+  bodyProperties: [
+    'agreedCaseWithManager',
+    'managerName',
+    'agreementDate',
+    'agreementDate-year',
+    'agreementDate-month',
+    'agreementDate-day',
+    'agreementSummary',
+  ],
+})
+export default class ExceptionDetails implements TasklistPage {
+  title = 'Provide details for exemption application'
+
+  questions = {
+    agreedCaseWithManager: 'Have you agreed the case with a senior manager?',
+    managerName: 'Name of senior manager',
+    agreementDate: 'What date was this agreed?',
+    agreementSummary: 'Provide a summary of the reasons why this is an exempt application',
+  }
+
+  body: ExceptionDetailsBody
+
+  constructor(_body: Partial<ExceptionDetailsBody>) {
+    this.body = {
+      agreedCaseWithManager: _body.agreedCaseWithManager,
+      managerName: _body.managerName,
+      'agreementDate-year': _body['agreementDate-year'],
+      'agreementDate-month': _body['agreementDate-month'],
+      'agreementDate-day': _body['agreementDate-day'],
+      agreementDate: DateFormats.dateAndTimeInputsToIsoString(
+        _body as ObjectWithDateParts<'agreementDate'>,
+        'agreementDate',
+      ).agreementDate,
+      agreementSummary: _body.agreementSummary,
+    }
+  }
+
+  response() {
+    let response = { [this.questions.agreedCaseWithManager]: sentenceCase(this.body.agreedCaseWithManager) }
+
+    if (this.body.agreedCaseWithManager === 'yes') {
+      response = {
+        ...response,
+        [this.questions.managerName]: this.body.managerName,
+        [this.questions.agreementDate]: DateFormats.isoDateToUIDate(this.body.agreementDate),
+        [this.questions.agreementSummary]: this.body.agreementSummary,
+      }
+    }
+
+    return response
+  }
+
+  previous() {
+    return 'is-exceptional-case'
+  }
+
+  next() {
+    return 'sentence-type'
```

Comment:
Is any work needed on the `previous()` method for the sentence-type page, to conditionally go back to the exception details page?

Reply:
Oh good catch!